### PR TITLE
Changed sharps to music sharps unicode

### PR DESCRIPTION
**Pull request by mvoorhes**
Took a suggestion from a comment on my last pull request #6865 that suggested the sharps signs be turned into the unicode versions of the music sharp signs rather than using # . Here is what it looks like now.

Screenshot 2023-09-17 at 6 34 09 PM

### CODE REVIEW

**Review by Veratil — APPROVED** [review]

**Review by zonkmachine — APPROVED** [review]

**Comment by zonkmachine** [comment]
Please edit the commit message on merge.

**Comment by superpaik** [comment]
With MSVC build those unicode symbols are not visible (you get a ?) Tested with local build and also with botbuild. It works with mingw, though.
This string is also used to get the names of the notes inside pianoroll, with the function "getNoteString"
While it makes sense in the drop-down, on the notes I think it is not a good idea to have both names.

File: src/gui/editors/PianoRoll.cpp
Comment (superpaik) [inline src/gui/editors/PianoRoll.cpp:133]: could be static const

**Comment by sakertooth** [comment]
> Please edit the commit message on merge.
I think the title is fine. I cannot think of a better name. Any suggestions?

**Comment by zonkmachine** [comment]
> > Please edit the commit message on merge.
> 
> I think the title is fine. I cannot think of a better name. Any suggestions?

No, the title is fine. I'm just thinking it's a good thing to edit out the automatic merge message since there is four commits in there and only the title is really necessary here.

File: src/gui/editors/PianoRoll.cpp
Comment (DomClark) [reply src/gui/editors/PianoRoll.cpp:133]: I agree, but it was already non-const, so I don't think it needs to be changed in this PR. My concern would be more with the 186-character line - it could do with being broken in a few places.

**Comment by DomClark** [comment]
> With MSVC build those unicode symbols are not visible

It sounds like MSVC is using the wrong character sets. I'll open a separate PR to fix this; I don't think it's a reason to hold off merging this one.

File: src/gui/editors/PianoRoll.cpp
Comment (Veratil) [reply src/gui/editors/PianoRoll.cpp:133]: If we do want a style fixup:
```suggestion
static std::array<QString, 12> s_noteStrings {
	"C", "C\u266F / D\u266D", "D", "D\u266F / E\u266D", "E", "F", "F\u266F / G\u266D", "G", "G\u266F / A\u266D", "A", "A\u266F / B\u266D", "B"
};
```

**Review by sakertooth — APPROVED** [review]

**Comment by superpaik** [comment]
> > With MSVC build those unicode symbols are not visible
> 
> It sounds like MSVC is using the wrong character sets. I'll open a separate PR to fix this; I don't think it's a reason to hold off merging this one.

Could be. It can be also because of QT. If the unicode is introduced by QChar it works ok, like QChar(0x266F).

**Comment by superpaik** [comment]
The other thing I commented (names of notes on pianoroll), i think should be adressed before mergin, because the note name is hidden if the string is "bigger" than the note, so it create some visual inconsistencies (see image)
Captura

**Comment by DomClark** [comment]
From the MSVC build log (https://github.com/LMMS/lmms/actions/runs/6217215088/job/16872023566#step:9:317):
```
D:\a\lmms\lmms\src\gui\editors\PianoRoll.cpp(133): warning C4566: character represented by universal-character-name '\u266F' cannot be represented in the current code page (1252)
D:\a\lmms\lmms\src\gui\editors\PianoRoll.cpp(133): warning C4566: character represented by universal-character-name '\u266D' cannot be represented in the current code page (1252)
```
It looks like MSVC is using CP-1252. Building with the `/utf-8` flag fixes the issue, so I will make a pull request to add it this evening.
> Could be. It can be also because of QT. If the unicode is introduced by QChar it works ok, like QChar(0x266F).

This is because 0x266F is a number, so is unaffected by the choice of character encodings. Qt interprets it as UTF-16 at runtime, at which point any interference from the compiler is long out of the way.

**Comment by sakertooth** [comment]
@mvoorhes can the format on line 133 be fixed before merge?

File: src/gui/editors/PianoRoll.cpp
Comment (mvoorhes) [inline src/gui/editors/PianoRoll.cpp:133]: Changed format.

**Review by DomClark — APPROVED** [review]

**Pull request merged**
